src/coord/src/coord.rs

```diff
@@ -320,20 +291,25 @@ pub struct Coordinator {
     /// active connections.
     active_conns: HashMap<u32, ConnMeta>,
 
-    /// Holds pending compaction messages to be sent to the dataflow workers. When
-    /// `since_handles` are advanced or `txn_reads` are dropped, this can advance.
-    index_since_updates: Rc<RefCell<HashMap<GlobalId, Antichain<Timestamp>>>>,
-    /// Holds pending compaction messages to be sent to the dataflow workers. When
-    /// `since_handles` are advanced or `txn_reads` are dropped, this can advance.
-    source_since_updates: Rc<RefCell<HashMap<GlobalId, Antichain<Timestamp>>>>,
-    /// Holds handles to ids that are advanced by update_upper.
-    since_handles: HashMap<GlobalId, AntichainToken<Timestamp>>,
-    /// Tracks active read transactions so that we don't compact any indexes beyond
-    /// an in-progress transaction.
-    // TODO(mjibson): Should this live on a Session?
-    txn_reads: HashMap<u32, TxnReads>,
-    /// Tracks write frontiers for active exactly-once sinks.
-    sink_writes: HashMap<GlobalId, SinkWrites<Timestamp>>,
+    /// For each identifier, its read policy and any transaction holds on time.
+    ///
+    /// The antichain contains the accumulation of all expressed pins, and prevents
+    /// compaction past its lower bound, to ensure they all remain valid. The antichain
+    /// is emptied only when all transactions have dropped, at which point the read
+    /// policy for the identifier is returned to its current compaction policy.
+    read_capability_needs: HashMap<GlobalId, ReadCapabilityState<mz_repr::Timestamp>>,
+    /// For each transaction, the pinned storage and compute identifiers and time at
+    /// which they are pinned.
+    ///
+    /// Upon completing a transaction, this timestamp should be removed from the holds
+    /// in `self.read_capability_needs[id]`.
+    txn_reads_neu: HashMap<u32, TxnReads>,
+    /// Sources to reconsider for read capability downgrades.
+    ///
+    /// This exists primarily to allow the coordinator to allow `persist` to compact
+    /// its collections, which is not handled through the STORAGE abstraction at the
+    /// moment. Should that change, this monitoring can be removed.
+    storage_compaction_opportunities: HashSet<GlobalId>,
```

Comment:
This could be deleted, and `fn maintenance()` moved in to its one call site (responses to timestamp bindings).